Docs: remove/warn about references to FOSUSerBundle.

m-ar-c opened the pull request:
## Subject

Update documentation because `FOSUSerBundle` is not used by `SonataUserBundle` anymore.


I am targeting this branch, because the change is BC.

Commit:
Docs: remove/warn about references to FOSUSerBundle.

Update documentation because FOSUSerBundle is not used by
SonataUserBundle anymore.

VincentLanglet commented:
HI, I removed the changelog, there is no need for doc update

VincentLanglet commented on docs/reference/security.rst:
```diff
@@ -6,14 +6,13 @@ User management
 
 By default, the SonataAdminBundle does not come with any user management,
 however it is most likely the application requires such a feature. The Sonata
-Project includes a ``SonataUserBundle`` which integrates the ``FOSUserBundle``.
+Project includes a ``SonataUserBundle``.
```
Maybe we can do better like
```
By default, the SonataAdminBundle does not come with any user management,
however it is most likely the application requires such a feature. For this, you can try
the ``SonataUserBundle``.
```

jordisala1991 commented on docs/reference/security.rst:
```diff
@@ -252,6 +251,11 @@ And specify it as Sonata security handler on your configuration:
 ACL and FriendsOfSymfony/UserBundle
 -----------------------------------
 
+.. note::
```
To me we should remove all references to FOSUserBundle, and replace with references to SonataUserBundle when needed.

References to the install of SonataUserBundle should be pointed to the SonataUserBundle docs.

VincentLanglet commented on docs/reference/security.rst:
```diff
@@ -252,6 +251,11 @@ And specify it as Sonata security handler on your configuration:
 ACL and FriendsOfSymfony/UserBundle
 -----------------------------------
 
+.. note::
+
+   Some parts of this section are obsolete : ``SonataUserBundle`` does not use
+   ``FOSUserBundle`` anymore.
+
```
Let's update the doc instead of adding a note :)

Commit:
Docs: Replace all remaining references to FOSUSerBundle.

- Link to SonataUserBundle documentation when appropriate.
- Update documentation to set up ACL with SonataUserBundle, not
FOSUSerBundle.

m-ar-c committed:
Merge branch 'sonata-project:4.x' into docs/remove-FOSUserBundle

m-ar-c commented:
I think I did all that was suggested here. I have almost zero knowledge and experience with the things I edited, so please review my contribution carefully :)

Commit:
Docs: Change firewall name from main to admin.

Changed firewall name from main to admin, in order to be consistent
with the SonataUserBundle documentation.

Commit:
Merge branch 'docs/remove-FOSUserBundle' of github.com:m-ar-c/SonataAdminBundle into docs/remove-FOSUserBundle

jordisala1991 commented on docs/reference/security.rst:
```diff
@@ -339,22 +311,22 @@ and then configure SonataAdminBundle:
                 # ...
 
-In your ``config/packages/fos_user.yaml`` you then need to put the following:
+In your ``config/packages/sonata_user.yaml`` you then need to put the following:
```
I think this whole block can be removed

m-ar-c replied:
Sorry I'm not sure what block you're talking about. I should remove line 314 to 325 (the `config/packages/sonata_user.yaml` code block) ?

jordisala1991 replied:
Yes, exactly! Can you do it?

m-ar-c replied:
Np, I'll do it tomorrow or this week-end.

Commit:
docs: remove section about sonata_user.yaml configuration.

jordisala1991 commented on docs/reference/security.rst:
```diff
@@ -738,15 +697,15 @@ User list customization
     protected function getAclUsers(): \Traversable
     {
-        $userManager = $container->get('fos_user.user_manager');
+        $userManager = $container->get('sonata_user.user_manager');
```
```suggestion
        $userManager = $container->get('sonata.user.manager.user');
```

jordisala1991 committed:
Update docs/reference/security.rst

jordisala1991 commented:
Thank you @m-ar-c
